src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -375,6 +376,38 @@ public static ImmutableArray<T> WhereAsArray<T>(this ImmutableArray<T> array, Fu
             return false;
         }
 
+        public static async Task<bool> AnyAsync<T>(this ImmutableArray<T> array, Func<T, Task<bool>> predicateAsync)
+        {
+            int n = array.Length;
+            for (int i = 0; i < n; i++)
```

Comment:
why not a foreach?

Reply:
Copied from the method above this one.

Reply:
Is this a method that could benefit from `ValueTask<T>`?

Reply:
`async Task<bool>` already optimizes the case where the method returns synchronously.

Reply:
CC. @stephentoub

Reply:
Didn't realize that about `Task<bool>`. At the same time is there any downside to `ValueTask<bool>` here? Seems like it gives the API a more explicit declaration about it's intended behavior.

Reply:
> At the same time is there any downside to ValueTask<bool> here?

The downside is just the same as what's outlined in https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/.  If all consumers do is `await BlahAsync(...)`, then the downsides are limited; it's slightly faster to await an already completed `Task<bool>` than a `ValueTask<bool>`, and a state machine for code containing that await will be slightly larger if it's `ValueTask`, but we're talking micro differences that don't matter in the majority of cases.  If you expect the returned task to ever be used in other consumption patterns (e.g. stored to await later, multiple awaiters, cached lazy async operation for later consumption, potentially thrown away rather than awaited, etc.), there are big downsides to `ValueTask`.